### PR TITLE
Huobi fetch deposits/withdrawals

### PR DESCRIPTION
**Pull request by lottaeouss**
Implemented suggested changes for a previous PR as below
https://github.com/ccxt/ccxt/pull/4391

### CODE REVIEW

File: js/huobipro.js [hunk 1]
```diff
@@ -1017,4 +1019,154 @@ module.exports = class huobipro extends Exchange {
...
+                'rate': undefined,
+            },
+        };
+    }
+
+    parseTransactionStatusByType (status, type = undefined) {
+        if (type === undefined) {
+            return status;
+        }
+        let statuses = {
```
Comment (tankakatan) [inline js/huobipro.js]: There's no use to split statuses by type – they are either unique or semantically identical. So i suggest to make it `parseTransactionStatus` (without `ByType`) and get rid of the `type` argument.

File: js/huobipro.js [hunk 2]
```diff
@@ -1017,4 +1019,154 @@ module.exports = class huobipro extends Exchange {
...
+            'deposit': {
+                'unknown': 'pending',
```
Comment (tankakatan) [inline js/huobipro.js]: I think `unknown` is more like `failed`. Why do you think it should be interpreted as `pending`?

File: js/huobipro.js [hunk 3]
```diff
@@ -1017,4 +1019,154 @@ module.exports = class huobipro extends Exchange {
...
+                'reject': 'failed',
+                'pass': 'pending',
+                'wallet-reject': 'failed',
+                'confirmed': 'ok',
+                'confirm-error': 'failed',
+                'repealed': 'failed',
+                'wallet-transfer': 'pending',
+                'pre-transfer': 'pending',
+            },
+        };
```
Comment (tankakatan) [inline js/huobipro.js]: I have a few more statuses in my homegrown implementation of `parseTransactionStatus` some of them may be out of date but some of them are still mentioned in the [docs](https://github.com/huobiapi/API_Docs_en/wiki/REST_Reference#withdraw-states). So could you please check it one more time:

```javascript
const statuses = {
        // deposit statuses
        'unknown': 'failed',
        'confirming': 'pending',
        'confirmed': 'ok',
        'safe': 'ok',
        'orphan': 'failed',
        // withdrawal statuses
        'submitted': 'pending',
        'reexamine': 'pending',
        'canceled': 'canceled',
        'pass': 'ok',
        'reject': 'failed',
        'pre-transfer':	'pending',
        'wallet-transfer': 'pending',
        'wallet-reject': 'failed',
        // 'confirmed': 'ok', // already present among deposit statuses
        'confirm-error': 'failed',
        'repealed': 'canceled',
        'valid': 'ok',
    }
```

File: js/huobipro.js (hunk 2 above)
Comment (lottaeouss) [reply js/huobipro.js]: I agree

File: js/huobipro.js (hunk 3 above)
Comment (lottaeouss) [reply js/huobipro.js]: Checked the docs and updated the statuses accordingly thanks

**Comment by lottaeouss** [comment]
@tankakatan Thanks for the feedback, I have implemented the suggested changes. Cheers

**Comment by tankakatan** [comment]
> @tankakatan Thanks for the feedback, I have implemented the suggested changes. Cheers

Well done. I have no other comments on this PR. @kroitor maybe you have something to add? If not, we are free to merge this.

**Pull request merged**